GliaWidgets/SecureConversations/SecureConversations.Availability.swift

```diff
@@ -0,0 +1,50 @@
+import Foundation
+
+extension SecureConversations {
+    struct Availability {
+        typealias CompletionResult = (Result<Bool, CoreSdkClient.SalemoveError>) -> Void
+        var environment: Environment
+
+        func checkSecureConversationsAvailability(completion: @escaping CompletionResult) {
+            environment.listQueues { queues, error in
+                if let error = error {
+                    completion(.failure(error))
+                    return
+                }
+
+                self.checkQueues(queues, completion: completion)
+            }
+        }
+
+        private func checkQueues(
+            _ queues: [CoreSdkClient.Queue]?,
+            completion: (Result<Bool, CoreSdkClient.SalemoveError>) -> Void
+        ) {
+            guard let queues = queues, isSecureConversationsAvailable(in: queues) else {
+                completion(.success(false))
+
+                return
+            }
+
+            completion(.success(true))
+        }
+
+        private func isSecureConversationsAvailable(
+            in queues: [CoreSdkClient.Queue]
+        ) -> Bool {
+            let filteredQueues = queues
+                .filter { self.environment.queueIds.contains($0.id) }
+                .filter { $0.state.status != .closed }
+                .filter { $0.state.media.contains(CoreSdkClient.MediaType.messaging) }
+
+            return !filteredQueues.isEmpty
+        }
+    }
+}
+
+extension SecureConversations.Availability {
+    struct Environment {
+        var listQueues: CoreSdkClient.ListQueues
+        var queueIds: [String]
```

Comment:
do we need to have the possibility to change these properties in runtime or could we make them as `let`?

Reply:
Fixed fbb0d8ef7636a0daeeaf95d166b4084fca2f42f9